src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -461,6 +461,11 @@ public bool TryGetPendingTransaction(Keccak hash, out Transaction transaction)
             return transaction != null;
         }
 
+        public bool IsTransactionKnown(Keccak hash)
+        {
+            return _hashCache.Get(hash);
```

Comment:
Now good question. Is know would be hashCashe? Or maybe we should only check actual held transactions? Or maybe we should also check DB?

Reply:
hash cache is better because we have heard about the txs from the hash cache

Reply:
we can do two-level ratio though

Reply:
added checking actual held transactions and needed to move whole logic from OnChainTxWatcher to BlockProcessor (in OnChainTxWatcher an attempt to remove transactions always returns false - they are already removed)